app/assets/javascripts/discourse/app/instance-initializers/highlight-posts.js

```diff
@@ -0,0 +1,19 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+export default {
+  initialize() {
+    withPluginApi("1.36.0", (api) => {
+      const username = api.getCurrentUser()?.username;
+      if (!username) {
+        return;
+      }
+
+      const style = document.createElement("style");
+      style.type = "text/css";
+      const cssRule = `.inline-onebox[data-author="${username}"] { background: var(--tertiary-400); }`;
+      style.appendChild(document.createTextNode(cssRule));
+
+      document.head.appendChild(style);
```

Comment:
@davidtaylorhq I would like your expertise on what you think of this solution please.

I see we already do this in `app/assets/javascripts/discourse/app/instance-initializers/hashtag-css-generator.js` for example.

I have found cases where people recommend to wrap the `appendChild` in a RAF call, but couldn't see any difference when looking at performance of this with/without in chrome.

Should we have an API to append a custom style only once?

Reply:
I'd recommend adding it to the `<DStyles />` component. That way, there's no manual messing with HTML, and cleanup is handled automatically.

We should do the same for hashtag-css-generator. There's no cleanup, so it's a memory leak in tests right now.

Taking a step back though: I don't think it's a good idea to implement this as a dynamic selector in a `<style>` tag. It makes the styling almost impossible for themes to override/customize. Instead, I think if we should add a decorateCooked which adds a class like `.current-user-post-link` to relevant links. Then the actual styling can be in a regular CSS file, and can easily be customized/disabled by themes.

That would also line up better with the self-mention styling implementation.

Reply:
Ok I didn't know DStyles :+1:

Im fine with the `decorateCooked` but actually tried to avoid it as I thought this solution was more efficient.